File: core/fgo/factor/position_factor.m
classdef position_factor < factor
    properties

    end
    
    methods
        function obj = position_factor(states, z, Omega)
            obj@factor(states,z,Omega);
        end

        function obj = evaluate(obj)
            % Extract states
            state = obj.states(1);

            % Position factor
            obj.A = -eye(4);
            obj.b = obj.z -  state.value;

            
            % (TODO): Mutiply the obj.Omega to A
            L = chol(obj.Omega, 'lower');
            obj.A = L' * obj.A;            
            obj.b = L' * obj.b;

        end
    end
end

File: core/fgo/factor/position_factor.py
import numpy as np
from .factor import Factor, whiten


class PositionFactor(Factor):
	def evaluate(self):
		self.A, self.b = whiten(-np.eye(4), self.z - self.states[0].value, self.omega)
		return self

__all__ = ["PositionFactor"]
